From 2440d1b96b7af95259867b22ba8b4ba677ef0ade Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Fri, 8 Nov 2024 17:43:07 +0000
Subject: [PATCH] DevicesModel: Register QList of QStringList to DBus

Avoids an abort when adding a new printer from kcmshell.


(cherry picked from commit c80a5a0d5e59049c2007264a41228fcfbaeda58b)

Co-authored-by: Kai Uwe Broulik <kde@privat.broulik.de>
---
 src/libkcups/DevicesModel.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libkcups/DevicesModel.cpp b/src/libkcups/DevicesModel.cpp
index 3ffc90673..8498a719e 100644
--- a/src/libkcups/DevicesModel.cpp
+++ b/src/libkcups/DevicesModel.cpp
@@ -45,6 +45,7 @@ DevicesModel::DevicesModel(QObject *parent)
 
     qDBusRegisterMetaType<MapSS>();
     qDBusRegisterMetaType<MapSMapSS>();
+    qDBusRegisterMetaType<QList<QStringList>>();
 
     // Adds the other device which is meant for manual URI input
     insertDevice(QLatin1String("other"), QString(), i18nc("@item", "Manual Config"), QString(), QLatin1String("other"), QString());
-- 
GitLab

